src/fastga_he/models/propulsion/components/tanks/gaseous_hydrogen_tank/components/sizing_tank_length.py

```diff
@@ -0,0 +1,185 @@
+            units="m**3",
+            val=np.nan,
+            desc="Capacity of the tank in terms of volume",
+        )
+
+        self.add_input(
+            name="data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":dimension:inner_diameter",
+            units="m",
+            val=np.nan,
+            desc="Inner diameter of the gaseous hydrogen tanks",
+        )
+
+        self.add_input(
+            name="data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":dimension:outer_diameter",
+            units="m",
+            val=np.nan,
+            desc="Outer diameter of the gaseous hydrogen tanks",
+        )
+        if in_fuselage:
+            self.add_input(
+                "data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+                + gaseous_hydrogen_tank_id
+                + ":number_of_tank",
+                val=1.0,
+                desc="Number of gaseous hydrogen tank in a stack in fuselage. "
+                "Default set 1.0 for single tank in fuselage and outside fuselage uses.",
+            )
+
+        self.add_output(
+            "data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":dimension:length",
+            val=1.0,
+            units="m",
+            desc="Value of the cylindrical length of the tank in the x-direction, "
+            "computed differently based on the location of the tank.",
+        )
+
+        self.declare_partials(of="*", wrt="*", method="exact")
+
+        self.declare_partials(
+            of="*",
+            wrt="data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":dimension:outer_diameter",
+            val=1.0,
+        )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        gaseous_hydrogen_tank_id = self.options["gaseous_hydrogen_tank_id"]
+        position = self.options["position"]
+        in_fuselage = not (position == "wing_pod" or position == "underbelly")
+
+        d_inner = inputs[
+            "data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":dimension:inner_diameter"
+        ]
+
+        d_outer = inputs[
+            "data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":dimension:outer_diameter"
+        ]
+
+        inner_volume = inputs[
+            "data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+            + gaseous_hydrogen_tank_id
+            + ":inner_volume"
+        ]
+        if in_fuselage:
+            nb_tank = inputs[
+                "data:propulsion:he_power_train:gaseous_hydrogen_tank:"
+                + gaseous_hydrogen_tank_id
+                + ":number_of_tank"
+            ]
+        else:
+            nb_tank = 1.0
```

Comment:
Instead of using the variable associated with the number of tanks everywheren wouldn't it be easier to have a component that compute the volume of *1* tank and then do the computation of all dimension and weight for *1* tank and then when you need the weight you re-multiply by the number of tanks. Just a suggestion